src/rules/jsdocFormatRule.ts

```diff
@@ -40,44 +42,172 @@ export class Rule extends Lint.Rules.AbstractRule {
         options: {
             type: "array",
             minItems: 0,
-            maxItems: 1,
+            maxItems: 3,
             items: {
                 type: "string",
-                enum: [OPTION_CHECK_MULTILINE_START],
+                enum: [OPTION_CHECK_MULTILINE_START, OPTION_PREFER_SINGLELINE],
             },
         },
-        optionExamples: [true, [true, OPTION_CHECK_MULTILINE_START]],
+        optionExamples: [
+            true,
+            [true, OPTION_CHECK_MULTILINE_START],
+            [true, OPTION_PREFER_SINGLELINE, { OPTION_MAX_WIDTH: 80}],
+            [true, OPTION_CHECK_MULTILINE_START, OPTION_PREFER_SINGLELINE, { OPTION_MAX_WIDTH: 80 }],
+        ],
         type: "style",
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Keep short jsdoc comments to a single line",
+                config: Lint.Utils.dedent`
+                    "rules": { "jsdoc-format": [true, "prefer-singleline", {"max-width": 80}] }
+                `,
+                pass: Lint.Utils.dedent`
+                    /**
+                     * good multline
+                     * jsdoc docblock
+                     */
+
+                    /** good singleline jsdoc dockblock */
+                `,
+                fail: Lint.Utils.dedent`
+                    /** bad single line jsdock block because it's way too long to fit on a single line */
+
+                    /**
+                     * bad multiline jsdoc docblock
+                     */
+                `,
+            },
+        ],
     };
     /* tslint:enable:object-literal-sort-keys */
 
     public static ALIGNMENT_FAILURE_STRING = "asterisks in jsdoc must be aligned";
     public static FORMAT_FAILURE_STRING = "jsdoc is not formatted correctly on this line";
+    public static PREFER_SINGLE_LINE_FAILURE = "short jsdoc block should be on single line";
+    public static SINGLE_LINE_MAX_WIDTH_FAILURE = "singleline jsdoc block should fit within maxWidth characters";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const maxWidthOption = this.ruleArguments.find((arg) => typeof arg === "object" && typeof arg[OPTION_MAX_WIDTH] === "number");
+        let maxWidth;
+        if (maxWidthOption) {
+            maxWidth = maxWidthOption[OPTION_MAX_WIDTH];
+        }
+
         return this.applyWithFunction(sourceFile, walk, {
             firstLineOfMultiline: this.ruleArguments.indexOf(OPTION_CHECK_MULTILINE_START) !== -1,
+            maxWidth,
+            preferSingleline: this.ruleArguments.indexOf(OPTION_PREFER_SINGLELINE) !== -1,
         });
     }
 }
 
 interface Options {
     firstLineOfMultiline: boolean;
+    preferSingleline: boolean;
+    maxWidth: number;
+}
+
+function shouldBeSingleline(ctx: Lint.WalkContext<Options>, lines: string[], {pos, end}: ts.CommentRange): boolean {
+    if (!ctx.options.preferSingleline) {
+        return false;
+    }
+
+    if (lines.length === 3) {
```

Comment:
is the max width not taken into account here?

Reply:
As far as I can tell, it doesn't need to be. This function will fix it to a single-line docblock and then `shouldBeMultiline` will fix it back to multiline and break it up appropriately if fitting it on a single line is inappropriate.